### DISCUSSION TITLE
Showing the parameter names for own tasks in block coding would be great

### DISCUSSION THREAD

**Discussion by jayyellow86** [pybricks/support#1582] [Ideas]
Hi,
my son i using the block coding with his wro team and it is awesome.
To structure their code, they use own tasks with names like "follow the line" or "drive until white".
These blocks usually have parameters like "speed" or "distance" or something similar.
The problem is, that you have to remember the parameters, because you don't have an indication of the parameter name when using this blocks. For example the Block is defined like "follow the line (distance) (speed) (factor)", when used it reads "follow the line ( ) ( ) ( )" with no indication of the parameter names.
It would be great, if the parameter names could be shown, when you use a own tasks. Ideally the name is always shown, even if the parameter is set to a given value. This would help the kids to understand their coding much easier.

Nice to have would be the option, to add additional text between parameters.
Then blocks could be like "follow the line for ( distance = 100 ) mm with ( speed = 200 ) °/s".
This would be great for younger children to easily read what is happening.

Kind Regards
Julian

**Comment by laurensvalk** [comment]
Thanks Julian! I'm glad you're enjoying Pybricks for WRO :)

Your requested feature would definitely be nice to have. We're collecting some ideas for this in https://github.com/pybricks/support/issues/1531

Feel free to add additional suggestions there too.
